sys/include/suit/coap.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2019 Kaspar Schleiser <kaspar@schleiser.de>
+ *               2019 Inria
+ *               2019 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+/**
+ * @defgroup    sys_suit SUIT secure firmware updates
+ * @ingroup     sys
+ * @brief       SUIT secure firmware updates
+ *
+ * @{
+ *
+ * @brief       SUIT CoAP helper API
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ */
+
+#ifndef SUIT_COAP_H
+#define SUIT_COAP_H
+
+#include "net/nanocoap.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief    Start SUIT CoAP thread
+ */
+void suit_coap_run(void);
+
+/**
+ * @brief SUIT CoAP endpoint entry.
+ *
+ * In order to use, include this header, then add SUIT_COAP_SUBTREE to the nanocoap endpoint array.
+ * Mind the alphanumerical sorting!
+ *
+ * See examples/suit_update for an example.
+ */
+#define SUIT_COAP_SUBTREE \
+    { \
+        .path="/suit/", \
+        .methods=COAP_MATCH_SUBTREE | COAP_METHOD_GET | COAP_METHOD_POST | COAP_METHOD_PUT, \
+        .handler=coap_subtree_handler, \
+        .context=(void*)&coap_resource_subtree_suit \
+    }
+
+
+/*
+ * Dear Reviewer,
+ *
+ * At the time of PR'ing this code, there was a pile of CoAP PR's waiting for
+ * reviews.  Some of that functionality is needed in one way or another for
+ * SUIT. In order to not block software updates with CoAP refactoring, some of
+ * the work-in-progress code has been copied here.  We expect this to be
+ * removed as soon as CoAP in master provides similar functionality.
+ *
+ * As this is internal code that will go soon, I exclude this from Doxygen.
+ *
+ * Kaspar (July 2019)
+ */
+#ifndef DOXYGEN
+
+/* subtree support */
+ssize_t coap_subtree_handler(coap_pkt_t *pkt, uint8_t *buf, size_t len,
```

Comment:
Some of the functions excluded from `doxygen` don't have an equivalent in the PR's I have seen for CoAP (might have missed it.):

- `coap_subtree_handler`
- `suit_coap_get_blockwise_url`

Reply:
yes. Let's consider thoe "internal functions".